be2-scala/src/main/scala/ch/epfl/pop/storage/DbActor.scala

```diff
@@ -951,7 +946,7 @@ object DbActor {
 
   /** Response for a [[ReadRumors]]
     */
-  final case class DbActorReadRumors(foundRumors: Map[PublicKey, List[Rumor]]) extends DbActorMessage
+  final case class DbActorReadRumors(foundRumor: Option[Rumor]) extends DbActorMessage
```

Comment:
If it returns only one rumor:
```suggestion
  final case class DbActorReadRumor(foundRumor: Option[Rumor]) extends DbActorMessage
```